plugin.audio.podcasts/addon.py

```diff
@@ -0,0 +1,359 @@
+
+        self.message = msg
+
+
+class Mediathek:
+
+    _GROUPS = 10
+    _ENTRIES = 10
+
+    _menu = None
+
+    addon_handle = None
+
+    def __init__(self):
+
+        groups = []
+
+        # build opml feeds
+        for g in range(self._GROUPS):
+
+            if settings.getSetting("opml_file_%i" % g) == "":
+                continue
+
+            path = os.path.join(
+                addon_dir, settings.getSetting("opml_file_%i" % g))
+            opml_data = self._load_opml(path)
+            try:
+                entries = []
+                for o in opml_data["opml"]["body"]["outline"]:
+                    if o["@type"] == "rss":
+                        entries += [{
+                            "path": "%i/%i" % (g, len(entries)),
+                            "name": o["@title"],
+                            "params": [
+                                {
+                                    "rss": o["@xmlUrl"]
+                                }
+                            ],
+                            "node": []
+                        }]
+
+                groups += [{
+                    "path": "opml-%i" % g,
+                    "name": opml_data["opml"]["head"]["title"],
+                    "node": entries
+                }]
+
+            except:
+                xbmc.log("Cannot parse opml file %s" % path, xbmc.LOGERROR)
+                pass
+
+        # build groups acc. settings
+        for g in range(self._GROUPS):
+
+            if settings.getSetting("group_%i_enable" % g) == "false":
+                continue
+
+            entries = []
+            for e in range(self._ENTRIES):
+
+                if settings.getSetting("group_%i_rss_%i_enable" % (g, e)) == "false":
+                    continue
+
+                icon = settings.getSetting("group_%i_rss_%i_icon"
+                                           % (g, e))
+
+                entries += [{
+                    "path": "%i" % e,
+                    "name": settings.getSetting("group_%i_rss_%i_name"
+                                                % (g, e)),
+                    "params": [
+                        {
+                            "rss": settings.getSetting("group_%i_rss_%i_url" % (g, e))
+                        }
+                    ],
+                    "icon": icon,
+                    "node": []
+                }]
+
+            groups += [{
+                "path": "pod-%i" % g,
+                "name": settings.getSetting("group_%i_name" % g),
+                "node": entries
+            }]
+
+        self._menu = [
+            {  # root
+                "path": "",
+                "node": groups
+            }
+        ]
+
+    def _play_latest(self, url):
+
+        try:
+            title, description, image, items = self._load_rss(url)
+            item = items[0]
+            li = self._create_playable_list_item(item)
+            xbmcplugin.setResolvedUrl(self.addon_handle, True, li)
```

Comment:
if you add this:

```
li.setProperty("IsPlayable", "true")
li.setProperty("path", item["stream_url"])
```

and remove 

``` 
xbmc.Player().play(item["stream_url"], li)
``` 

It should be as it is and Kodi should start playing the stream.

Reply:
@basrieter 
I added `li.setProperty("path", item["stream_url"])` in `_create_playable_list_item` although I have already initialized the ListItem with `path`, see `li = xbmcgui.ListItem(label=item["name"], label2=item["name2"], path=item["stream_url"])`. But it seems to make a difference. 

`li.setProperty("IsPlayable", "true")` was already there, see  `_create_playable_list_item`

Last but not least I removed `xbmc.Player().play(item["stream_url"], li)`

Reply:
@basrieter 
Actually I see now that `xbmc.Player().play(item["stream_url"], li)` is still required since the stream doesn't start if this is missing. Any idea?

Reply:
I guess it's something wrong with your routing logic. the playable listitem (the one with IsPlayable = True) should point to a "plugin://yourplugin/something" path and not to the final stream url. When that path is called, you then set the resolved url to the final path. The reason we suggest this approach is because it has implications on the database. This way your "path" is always the same (the plugin one) and you basically avoid adding new entries to the database everytime the stream changes (since it is dynamic)

Reply:
Instead of the `setProperty()` use `item.setPath()`. That should do the trick.

Reply:
Shall I waith then for you to test?